src/btree/bt_cursor.c

```diff
@@ -629,23 +629,19 @@ __wt_btcur_search_prepared(WT_CURSOR *cursor, WT_UPDATE **updp)
     WT_BTREE *btree;
     WT_CURSOR_BTREE *cbt;
     WT_UPDATE *upd;
-
-    *updp = NULL;
-
-    cbt = (WT_CURSOR_BTREE *)cursor;
-    btree = CUR2BT(cbt);
-    upd = NULL; /* -Wuninitialized */
+    bool leaf_found;
 
     /*
-     * Not calling the cursor initialization functions, we don't want to be tapped for eviction nor
-     * do we want other standard cursor semantics like snapshots, just discard the hazard pointer
-     * from the last operation. This also depends on the fact we're not setting the cursor's active
-     * flag, this is really a special chunk of code and not to be modified without careful thought.
+     * We don't need to know if we found the key on the ref we provide but the search function
```

Comment:
The previous version of this comment said the following: "this is really a special chunk of code and not to be modified without careful thought.". I have removed all logic it refers to which is admittedly a little concerning but I cannot determine what exactly we achieved by resetting the cursor here. One thing a reset does is it calls `__cursor_pos_clear`, I could in theory call that directly that way we keep our hazard pointer.

Before a `__cursor_reset` was added here a `__wt_cursor_func_init` was here which has additional semantics on top of the reset. I am unsure what exactly we need.

Reply:
From the ticket it appears we don't want the cursor to get trapped in eviction by avoiding setting the cursor's active flag.

Reply:
I could F_CLR the active flag here.

Reply:
I don't think this needs anything further to get the semantics originally desired. The change in WT-5480 followed soon after WT-5387. Before WT-5387, the code would open an entirely new cursor for each op being resolved, and the goal of this was to re-use cursors. With 5387 cursors were reused, but the could be co-opted into eviction between usages. 
OTOH: I'm concerned that this code is leaking hazard pointers somehow - we've seen some signal in the field that hazard pointers can be acquired in surprising ways..